### PR TITLE
Jamila Cornick Octos

### PR DESCRIPTION
**Pull request by Jcornick21**
# Hotel
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe a design decision you had to make when working on this project. What options were you considering? What helped you make your final decision? | I decided to make a class for rooms and also gave it a status. It seems like I didn tnecessarily need to. I probably would have changed that if I had more time to work on the project.
Describe a concept that you gained more clarity on as you worked on this assignment. | Igained more clarity around the idea that my administrator class would have access to everything but not everything would have access to administrator class
Describe a _nominal test_ that you wrote for this assignment. | I tested to see ifmy administrator class could access the total cost associated with my reservations.
Describe an _edge case test_ that you wrote for this assignment. | Didnt get to writing edge cases I was trying to finish nominal cases and code first 
How do you feel you did in writing pseudocode first, then writing the tests and then the code? | This process is great! very helpful.

### CODE REVIEW

File: lib/administrator.rb [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+module Hotel
+  class Administrator
+    attr_reader :room_list, :reservation_list
+
+    def initialize
+      @room_list = all_rooms
+      @reservation_list = []
+
+      if reservation_list == []
```
Comment (kariabancroft) [inline lib/administrator.rb:13]: How would this code path be executed where this conditional is false?

File: lib/administrator.rb [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+
+    # can see total cost of a reservation
+    def reservation_cost(res_name)
+      @reservation_list.each do |reservation|
+        if reservation.guest == res_name
+          res_cost = reservation.total_cost
+          # puts res_cost
+          return res_cost
+        elsif reservation.guest != res_name
```
Comment (kariabancroft) [inline lib/administrator.rb:63]: What will happen with this if the reservation you're looking for is the second in the list? (hint: unexpected behavior)

File: lib/reservation.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    end
+
+    def open_rooms(num_of_rooms)
+      reserved_units = []
+      admin = Hotel::Administrator.new
+      rm_units = admin.room_list
+      (num_of_rooms).times do
+        reserved_unit = rm_units.sample
+        if reserved_unit.status == :UNAVAILABLE
+          while reserved_unit.status == :UNAVAILABLE
```
Comment (kariabancroft) [inline lib/reservation.rb:37]: Why do you need an `if` and a `while`?

**Comment by kariabancroft** [comment]
# Hotel
## What We're Looking For

Feature | Feedback
---     | ---
**Design** |
Demonstrated classes having a single responsibility | Yes - I like the way you tracked the rooms with their status, this is definitely a challenge! 
Demonstrated loose coupling | Yes
Methods demonstrate a good use of encapsulation, inputs and outputs | Yes
Wave 1 requirements |Yes
Wave 2 requirements |Yes
Wave 3 requirements | No

More commits would be good!
